mergify_engine/rules/parser.py

```diff
@@ -14,103 +13,115 @@
 # WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 # License for the specific language governing permissions and limitations
 # under the License.
+import dataclasses
 import datetime
 import re
 import typing
 import zoneinfo
 
-import pyparsing
-
 from mergify_engine import date
 
 
-git_branch = pyparsing.CharsNotIn("~^: []\\")
-regexp = pyparsing.CharsNotIn("")
-positive_integer = pyparsing.Word(pyparsing.nums).setParseAction(
-    lambda toks: int(toks[0])
-)
-integer = pyparsing.Combine(
-    pyparsing.Optional(pyparsing.Literal("-")) + pyparsing.Word(pyparsing.nums)
-).setParseAction(lambda toks: int(toks[0]))
-github_login = pyparsing.Word(pyparsing.alphanums + "-[]")
-github_team = pyparsing.Combine(
-    pyparsing.Literal("@") + github_login + pyparsing.Literal("/") + github_login
-) | pyparsing.Combine(pyparsing.Literal("@") + github_login)
-text = (
-    pyparsing.QuotedString('"') | pyparsing.QuotedString("'") | pyparsing.CharsNotIn("")
-)
-milestone = pyparsing.CharsNotIn(" ")
+@dataclasses.dataclass
+class ConditionParsingError(Exception):
+    message: str
+
+
+conditions_parser = {
+    "number": "positive-number",
```

Comment:
What about using an `Enum` as values here?
Should be faster to compare against than strings.